### PR TITLE
Ship Keys

### PR DESCRIPTION
**Pull request by ZephyrTFA**
## About The Pull Request

Allows people to lock their helm console using a special key that is created at the helm console.

## Why It's Good For The Game

Allows Captains, or the first person to spawn a ship, to lock the Helm console to minimize the likelyhood of unsavory people doing unsavory things. Also gives tangible power to an individual over the ship.

## Changelog
:cl:
add: Helm Consoles can now be locked with a shipkey
add: Shipkeys, if your ship does not already have one , or was destroyed by an idiot, your helm console can print one
/:cl:

### CODE REVIEW

**Comment by MusselMain** [comment]
This seems like it'd also give the captains of vessels more incentive to avoid leaving their ship if they have locked the helm and are holding the keys, as if they die their crew would be unable to leave until they locate their corpse

**Comment by ZephyrTFA** [comment]
That would be a positive side effect, ideally captains are captains and not miners

**Comment by ilikesaying** [comment]
Can you make the console make those “beep-beep sounds when unlocked

File: code/modules/overmap/ships/controlled_ship_datum.dm [hunk 1]
```diff
@@ -238,3 +258,18 @@
 	shuttle_port.name = name
 	for(var/area/shuttle_area as anything in shuttle_port.shuttle_areas)
 		shuttle_area.rename_area("[name] [initial(shuttle_area.name)]")
+
+/obj/item/key/ship
+	name = "Ship Key"
```
Comment (Fikou) [inline code/modules/overmap/ships/controlled_ship_datum.dm]: object names are usually lowercase

**Comment by Vexylius** [comment]
bug:
* ship key name does not get updated when ship name is changed

possible issue:
* commonly appearing emag on syndicate ships is now a key opening all ships. That contributes to the problem where NT, SolGov and Independent ships are weaker than the syndicate
* if someone were to make the ship moving, then lock the console and walked out into hyperspace and died the crew would be doomed without admin intervention or https://github.com/shiptest-ss13/Shiptest/issues/1078

suggestions:
* would be cool if there was a way to let the AI pilot the ship despite console being locked
* could change the icon to something more visible, like a keycard maybe?
* could play /sound/machines/buzz_two.ogg when someone tries to use locked console
* could play /sound/machines/button3.ogg on unlocking and button4.ogg on locking
* if you could scan your captainship medal on the console or insert the key into the medal that would be cooler than to use a plain key
* ability to lock research server/ore silo
* remote locking/unlocking like on some car keys


overall a good feature

**Comment by ZephyrTFA** [comment]
If the key is destroyed, i.e. hyperspace, it automatically unlocks and makes a new key

**Comment by Vexylius** [comment]
it doesn't get destroyed when it's in the inventory of a corpse in hyperspace though, right?

**Comment by ZephyrTFA** [comment]
Touche

**Comment by ZephyrTFA** [comment]
You can now use a multi-tool to unlock a helm console, however doing so alerts the entire v level.
Remote unlocking is now possible by using the key in hand

**Comment by HanSolo1519** [comment]
May be worthwhile adding a sneakier hacking system to futureproof it against antags

IE deconstructing the helm and having a slow do_after of “Hotwiring” the circuit board that doesn't alert the crew, allowing for dastardly turncoats to hijack the ship while leaving their targets stranded planetside, while still leaving the tator very exposed during the whole process (Can't move without interrupting the hack, looks kinda suspicious to be standing over a deconstructed helm).

Not a demand, I'm unsure there ever will be antags in the future.

**Comment by Sylphily** [comment]
Sexy

**Pull request merged**